clinica/pipelines/t1_linear/t1_linear_utils.py

```diff
@@ -1,3 +1,32 @@
+from clinica.utils.exceptions import ClinicaParserError
+
+
+def extract_source_entities(ref_file):
```

Comment:
I feel like these functions are not specific to the `t1_linear` pipeline, they extract entities from bids file name and vice-versa. They could probably be shared across pipelines. WDYT?

Reply:
Indeed, that is the goal ultimately. This currently is sort of a POC to see if it works :)